Automattic/simplenote-android
# Add custom icon preference for sustainers

### Fix
This PR adds a new preference in the settings for sustainers to set a green app icon:

<img width="50%" src="https://github.com/Automattic/simplenote-android/assets/789137/36d87ac7-b4ae-4382-a9e3-a66a17ff6159.jpg" />
<img width="50%" src="https://github.com/Automattic/simplenote-android/assets/789137/edf1f3aa-8da6-4343-91d3-d4fc93e56883" />

Changing the icon is kind of a pain on Android involving `activity-alias` to use a different drawable for the (gulp) main activity. From my testing it is working pretty well but does crash the app when you swap out the activities. 

I also couldn't get the "New Note" shortcut to work, so just left that out when switching to the activity-alias.

### Test
* On a non-sustainer account, launch the app. You should see no sustainer icon setting in the preferences view and the app should work normally.
* Switch to a sustainer account, you can run this code to make yourself a sustainer:

```
long now = System.currentTimeMillis();
try {
    Preferences preferences = mPreferencesBucket.get(Preferences.PREFERENCES_OBJECT_KEY);
    preferences.setActiveSubscription(now);
} catch (BucketObjectMissingException e) {
    throw new RuntimeException(e);
}
```

* The sustainer icon preference should be in the settings.
* Flip the switch! The app should close and you'll see the green icon.
* Verify tapping the icon works and that you can still share content from other apps to Simplenote.

### Release
These changes do not require release notes.
